zeebe/dynamic-config/src/test/java/io/camunda/zeebe/dynamic/config/changes/StartPartitionScaleUpApplierTest.java

```diff
@@ -0,0 +1,223 @@
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.dynamic.config.changes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.dynamic.config.ClusterConfigurationAssert;
+import io.camunda.zeebe.dynamic.config.changes.ConfigurationChangeAppliers.ClusterOperationApplier;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.DynamicPartitionConfig;
+import io.camunda.zeebe.dynamic.config.state.MemberState;
+import io.camunda.zeebe.dynamic.config.state.PartitionState;
+import io.camunda.zeebe.dynamic.config.state.RoutingState;
+import io.camunda.zeebe.dynamic.config.state.RoutingState.MessageCorrelation.HashMod;
+import io.camunda.zeebe.dynamic.config.state.RoutingState.RequestHandling.ActivePartitions;
+import io.camunda.zeebe.dynamic.config.state.RoutingState.RequestHandling.AllPartitions;
+import io.camunda.zeebe.dynamic.config.state.RoutingState.RequestHandling.Mixed;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.test.util.asserts.EitherAssert;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+
+final class StartPartitionScaleUpApplierTest {
+  private final PartitionScalingChangeExecutor executor =
+      mock(PartitionScalingChangeExecutor.class);
+  private final ClusterConfiguration initialConfiguration =
+      ClusterConfiguration.init()
+          .addMember(
+              MemberId.from("1"),
+              MemberState.initializeAsActive(
+                  Map.of(
+                      1,
+                      PartitionState.active(1, DynamicPartitionConfig.init()),
+                      2,
+                      PartitionState.active(1, DynamicPartitionConfig.init()),
+                      3,
+                      PartitionState.active(1, DynamicPartitionConfig.init()))));
+
+  @Test
+  void shouldFailOnInvalidPartitionCounts() {
+    EitherAssert.assertThat(
+            new StartPartitionScaleUpApplier(executor, 0).init(initialConfiguration))
+        .left()
+        .asInstanceOf(InstanceOfAssertFactories.throwable(IllegalArgumentException.class))
+        .hasMessageContaining("Desired partition count must be greater than 1");
+
+    EitherAssert.assertThat(
+            new StartPartitionScaleUpApplier(executor, 100000).init(initialConfiguration))
+        .left()
+        .asInstanceOf(InstanceOfAssertFactories.throwable(IllegalArgumentException.class))
+        .hasMessageContaining("Desired partition count must not exceed 8192");
+  }
```

Comment:
:wrench: Would split this in two tests :shrug: There's no shared set up or anything anyway.